static/app.js

```diff
@@ -75,113 +189,161 @@ $(function() {
+        }
+        catch(e) {
+            console.error(e.message)
+        }
+    }
+
+    /** asynchronous function for searching a brewery based on an input term (city, name, state, type)
+     *  sends get request to API endpoint
+    */
+
+    async function termSearch(way_to_search, term) {
+        try {
+            const response = await axios({
+                url: `${base_url}/`,
+                method: "GET",
+                params: {
+                    [way_to_search]: term
                 }
-            }  
+            })
+            const breweries = processBreweries(response.data)
+            renderBreweries(breweries, '.results')
+        }   
+        catch(e) {
+            console.error(e.message)
         }
-        else {
-            $('.results').empty()
-            term = $('#term').val()
-            try {
-                const response = await axios({
-                    url: `${base_url}search/`,
-                    method: "GET",
-                    params: {
-                        query: term
-                    }
-                })
-                Object.entries(response.data).forEach((entry) => {
-                    const [key, value] = entry;
-                    const brewery = value
-                    renderBrewery(brewery)
-                })
-            }   
-            catch(e) {
-                console.log(e.message)
-            }
-        }    
-    })
+    }
 
-    function renderBrewery(brewery) {
-        $('#results-header').removeClass('d-none')
-        $('.results').append(
-            `<h5><a href='/breweries/${brewery.id}'>${brewery.name}</a></h5>
-            <p>${brewery.city}, ${brewery.state}</p>`)
-    
+    /** Mapping: uses google maps javascript API to render a map on brewery-details page */
+
+    /** create map variable */
+    let map;
+
+    /** asynchronous function to initialize map object 
+     * sets map location and marker based on latitude and longitude values present in the DOM 
+     * if either value is not present i.e. location, then displays to user that no map is available
+     */
+
+    async function initMap() {
+    const { Map } = await google.maps.importLibrary("maps");
+    let location = {lat: parseFloat($('#latitude').val()), lng: parseFloat($('#longitude').val())}
+    if (location.lat && location.lng) {
+        map = new Map($('#map')[0], {
+            center: location,
+            zoom: 12,
+        });
+        let marker = new google.maps.Marker({position:location, map:map})
+    }
+    else {
+        $('#map').addClass('d-flex flex-column border border-primary border-2 align-items-center justify-content-center')
+                .html(`<p>Unable to render map</p>
+                       <p>Coordinates not available</p>
+                    `)
+    }}
+
+    /** conditional that calls initMap when the window location is the brewery-detail page */
+
+    if (window.location.href.match(/\/breweries\/[a-f0-9-]+$/)) {
```

Comment:
maybe turn this into a utility function for searching up route destinations if you do this more than once in the future

Reply:
noted